mlir/lib/Dialect/Rock/utility/loweringUtils.cpp

```diff
@@ -19,11 +19,29 @@ using namespace mlir::rock;
 
 bool mlir::rock::isWrWAtomicKernel(GemmFeatures features, Type dataType,
                                    bool requiredPadding) {
-  return bitEnumContainsAll(features,
-                            GemmFeatures::mfma | GemmFeatures::atomic_add) &&
+  return mlir::rock::isAccel(features, dataType) &&
+         bitEnumContainsAll(features, GemmFeatures::atomic_add) &&
          (dataType.isF32() || dataType.isF16()) && !requiredPadding;
 }
 
+bool mlir::rock::isAccel(GemmFeatures features, Type dataType) {
+  bool isMfma = bitEnumContainsAll(features, GemmFeatures::mfma);
+  if (isMfma)
+    return true;
+
+  bool isWmma = bitEnumContainsAll(features, GemmFeatures::wmma);
+  if (!isWmma)
+    return false;
+
+  Type elementType = dataType;
+  if (auto vectorType = dataType.dyn_cast<VectorType>())
+    elementType = vectorType.getElementType();
+  else if (auto memrefType = dataType.dyn_cast<MemRefType>())
+    elementType = memrefType.getElementType();
+
+  return (elementType.isF16() || elementType.isInteger(8));
```

Comment:
The problem here is that I made so that wmma is by default on for gfx1100. Should I change this? I guess it makes sense, because the architecture won't be sufficient to determine what features are enabled. So to enable wmma the user should always need to ask for it. What do you think?

Reply:
I'd suggest that the function for default arch features grows a data type argument, or that we turn off WMMA for f32, or so on

Reply:
The data type argument was why I added  `isAccel(features, dataTYpe)`, so I can get the features + datatype and check if I can run the acceleator gemms. I guess @jerryyin is worried that those things might fail silently if we assume that the feature is on, but somehow the compiler decides to not use it. This is why I was thinking to turn the wmma feature off on gfx11 and then erroring if the user wants `wmma=on` for the wrong data type.

Reply:
So there's two separate issues here
1. Being able to control whether WMMA is on/off despite the defaults
2. That when the architecture supports wmma, that only applies for f16 and i8
3. We need to know what kind of gemm we'll be doing when we create the gemm so we can, for example, handle perf configs correctly

I claim that what we want to do here is that, instead of just looking up the architecture features in the arch db directly, we want a function that wraps around that that takes the arch name and the data type and masks off features that aren't relevant - specifically, f32 wmma.

Then, having wmma be `infer` works correctly, constructing GEMMs etc. from tosa works correctly, and so on ... _and_ if I decide to do `-t f32 -wmma=on` this'll generate that kernel (which will promptly trip on an error message but anyway)